### PR TITLE
feat: create `<VerticalNav />` component

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- Adds vertical nav component—simple, dense, and expandable variants
- Adds some icons in support of work
- Adds `2xs` icon size to tokens in order to support menu item arrows

## ✅ Checklist

- [x] Visuals are complete and match Figma
- [x] Code is complete and in accordance with our style guide
- [x] Design and theme tokens are audited for any relevant changes
- [x] Unit tests are written and passing
- [x] TSDoc is written or updated for any component API surface area
- [x] Ensure no accessibility violations are reported in Storybook
- [x] Specs and documentation are up-to-date
- [x] Changeset is added
- [x] Cross-browser check is performed (Chrome, Safari, Firefox)
- [x] Stories in Storybook accompany any relevant component changes

### CODE REVIEW

File: easy-ui-react/src/VerticalNav/Subnav.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const SubnavLevelContext = createContext<number>(0);
+
+/**
+ * Represents a nested navigation within a navigation item.
+ */
+export function Subnav(props: SubnavProps) {
+  const { selectedKey, ...listStateProps } = props;
+  const levelContext = useContext(SubnavLevelContext);
+  const level = useMemo(() => levelContext + 1, [levelContext]);
```
Comment (OskiTheCoder) [inline easy-ui-react/src/VerticalNav/Subnav.tsx:27]: **question** what is level capturing here?

File: easy-ui-react/src/VerticalNav/NavItem.module.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  color: theme-token("color.neutral.900");
+}
+
+.selectedTree {
+  color: theme-token("color.primary.600");
+}
+
+.selectedList {
+  color: theme-token("color.primary.500");
+}
```
Comment (OskiTheCoder) [inline easy-ui-react/src/VerticalNav/NavItem.module.scss:21]: **praise**: really like how our theme tokens are looking, these read much cleaner

File: .changeset/wet-camels-applaud.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@easypost/easy-ui-icons": minor
+---
+
+feat: add AwardStar, HomeWork, LocalPostOffice, Redeem, Timline icons
```
Comment (OskiTheCoder) [inline .changeset/wet-camels-applaud.md:5]: **question**: is there a benefit to having changesets on a per project basis?

Comment (stephenjwatkins) [reply .changeset/wet-camels-applaud.md:5]: the only benefit is having more specific messages for each project. i was previously lumping all projects into a single message like `create <x /> component` but when looking at the changes for a specific project by itself, they didn't make much sense

File: easy-ui-react/src/VerticalNav/Subnav.tsx (hunk 1 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/VerticalNav/Subnav.tsx:27]: level is capturing the nesting hierarchy for subnavigations that are within other subnavigations. the reason for this is that some styles are different depending on if it's a subnavigation or a subnavigation within a subnavigation.

let me know if that makes sense, or if there's an alternate name for this to better capture what it does

File: easy-ui-react/src/VerticalNav/NavItem.module.scss (hunk 2 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/VerticalNav/NavItem.module.scss:21]: i agree. more simple/straightforward

File: easy-ui-react/src/VerticalNav/Subnav.tsx (hunk 1 above)
Comment (OskiTheCoder) [reply easy-ui-react/src/VerticalNav/Subnav.tsx:27]: ah makes sense

**Comment by OskiTheCoder** [comment]
**question**: for the expandable variation, does it have a way to collapse and reopen again? or is the intent for the sub nav items to always display?

https://github.com/EasyPost/easy-ui/assets/44451175/03ba8f0a-6c89-4dff-b2f4-11f77a75d555

**Comment by stephenjwatkins** [comment]
> **question**: for the expandable variation, does it have a way to collapse and reopen again? or is the intent for the sub nav items to always display?

yeah it should normally collapse and reopen. VerticalNav is a controlled component, so it's left to the consumer to control when items are selected and expanded, and the stories in the storybook are all with fixed props for simplicity of the story file. i can add an example story, though, with state attached if you feel it would be helpful?

**Comment by OskiTheCoder** [comment]
> > **question**: for the expandable variation, does it have a way to collapse and reopen again? or is the intent for the sub nav items to always display?
> 
> yeah it should normally collapse and reopen. VerticalNav is a controlled component, so it's left to the consumer to control when items are selected and expanded, and the stories in the storybook are all with fixed props for simplicity of the story file. i can add an example story, though, with state attached if you feel it would be helpful?

ah got it. think its ok as is

**Review by OskiTheCoder — APPROVED** [review]
LGTM! really nice stuff here

**Pull request merged**